src/components/TodoModal/TodoModal.tsx

```diff
@@ -2,41 +1,69 @@
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
+import { currentTodoSlice } from '../../features/currentTodo';
 
 export const TodoModal: React.FC = () => {
+  const currentTodo = useAppSelector(state => state.currentTodo);
+  const dispatch = useAppDispatch();
+  const [user, setUser] = useState<User | null>(null);
+
+  if (!currentTodo) {
+    throw Error('Not todo');
+  }
```

Comment:
The error message 'Not todo' is not very descriptive. Consider providing a more informative error message, such as 'No current todo selected.'